### ISSUE TITLE
Allow to ignore Self-Signed SSL certificates

### ISSUE THREAD

**Issue by bararchy** [andresriancho/w3af#9071]
Scanning Local site with Self-Signed SSL: 

```

DNS response from DNS server for domain: 192.168.1.237
Incrementing global error count. GEC: 9
Failed to HTTP "GET" "https://192.168.1.237". Reason: "[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:581)", going to retry.
Traceback for this error: Traceback (most recent call last):
  File "/usr/share/w3af/core/data/url/extended_urllib.py", line 450, in _send
    res = self._opener.open(req)
  File "/usr/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/usr/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/usr/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/share/w3af/core/data/url/handlers/keepalive.py", line 790, in https_open
    return self.do_open(req)
  File "/usr/share/w3af/core/data/url/handlers/keepalive.py", line 616, in do_open
    self._start_transaction(conn, req)
  File "/usr/share/w3af/core/data/url/handlers/keepalive.py", line 751, in _start_transaction
    conn.endheaders()
  File "/usr/lib/python2.7/httplib.py", line 997, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 850, in _send_output
    self.send(msg)
  File "/usr/lib/python2.7/httplib.py", line 812, in send
    self.connect()
  File "/usr/lib/python2.7/httplib.py", line 1212, in connect
    server_hostname=server_hostname)
  File "/usr/lib/python2.7/ssl.py", line 350, in wrap_socket
    _context=self)
  File "/usr/lib/python2.7/ssl.py", line 566, in __init__
    self.do_handshake()
  File "/usr/lib/python2.7/ssl.py", line 788, in do_handshake
    self._sslobj.do_handshake()
SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:581)
```

**Comment by andresriancho** [comment]
The log line `Incrementing global error count. GEC: 9` tells me you're not using the latest w3af from master. Please upgrade and re-open the issue if it's still in the latest master

**Comment by secwineman** [comment]
Hello,

I'm using the latest W3af available in BlackArch's repositories and I'm experiencing the same issue.
After some digging I found that page on the Python Development Guide that describes the problem and gives a quick hack:
https://www.python.org/dev/peps/pep-0476/#opting-out
As far as I understand, Python now comes with a systematic verification of the ssl certs, which causes the issue. I modified the /usr/share/w3af/w3af/core/data/url/extended_urllib.py file with the quick hack provided in the url above and it worked. Hope it helped.
